lib/plugins/static.js

```diff
@@ -170,19 +171,19 @@ function serveStatic(options) {
         } else if (opts.appendRequestPath) {
             file = path.join(opts.directory, decodeURIComponent(req.path()));
         } else {
-            var dirBasename = path.basename(opts.directory);
-            var reqpathBasename = path.basename(req.path());
-
-            if (
-                path.extname(req.path()) === '' &&
-                dirBasename === reqpathBasename
-            ) {
+            if (path.extname(req.path()) === '') {
                 file = opts.directory;
             } else {
-                file = path.join(
-                    opts.directory,
-                    decodeURIComponent(path.basename(req.path()))
-                );
+                var fileName = decodeURIComponent(path.basename(req.path()));
+                // test for requested file, if it doesn't exist we send the directory as path since appendRequestPath = false
+                if (!fs.existsSync(path.join(opts.directory, fileName))) {
```

Comment:
nit: mind breaking these nested function invocations up into multiple lines w/ variables indicating what each path is supposed to represent?

Reply:
+1

Reply:
Sure! let me work on it and update the PR!